src/explorer.api/Models/DiffixApiClient.cs

```diff
@@ -164,6 +163,59 @@ async public Task<QueryResult<RowType>> PollQueryResult<RowType>(string queryId)
                 );
         }
 
+        async public Task<QueryResult<RowType>> PollQueryUntilComplete<RowType>(
+            string queryId,
+            CancellationToken ct,
+            TimeSpan? pollFrequency)
+        {
+            if (pollFrequency is null)
+            {
+                pollFrequency = TimeSpan.FromMilliseconds(500);
+            }
+
+            while (!ct.IsCancellationRequested)
+            {
+                var queryResult = await PollQueryResult<RowType>(queryId);
+                if (queryResult.Completed)
+                {
+                    return queryResult;
+                }
+                else
+                {
+                    await Task.Delay(pollFrequency.Value, ct);
+                }
+            };
+
+            ct.ThrowIfCancellationRequested();
```

Comment:
Shouldn't you cancel the query as well if you want to abort?

Reply:
Yes, in fact I should really cancel the underlying http request too (thought this is less of an issue). I made a note to do this I think (in the comments, possibly in a later file version...). Will file an Issue though so it isn't forgotten. 

(Done: issue #29)